lib/route.js

```diff
@@ -231,6 +232,12 @@ function buildRouting (options) {
         return
       }
 
+      // TODO: @MetCoder95 Implement hability for ignoring paths
+      if (opts.method === 'GET') {
+        const onSendHandlers = parseHeadOnSendHandlers(opts.onSend)
+        prepareRoute.call(this, 'HEAD', path, { ...opts, onSend: onSendHandlers })
```

Comment:
Did you evaluate to use `preSerialization`?

If the GET has a HUGE payload, it will be serialized with this implementation and then discarded.
Maybe could it be possible to avoid this process step?

Reply:
Hmm, actually didn't think about it but sounds logical to me. The only thing is that will not handle payloads where `streams`, `buffers` or `strings` are being sent (even `null`), for those cases `onSend` hook will still be needed, isn't it?
Maybe appending handlers for both hooks will cover both scenarios, what do you think?

Reply:
I don't think it's possible to avoid it. HEAD request should return the `content-length` header, and it's impossible to compute it _without_ serializing.